compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -233,19 +234,52 @@ OMR::ARM64::TreeEvaluator::lushrEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+// also handles lrolEvaluator
 TR::Register *
 OMR::ARM64::TreeEvaluator::irolEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::irolEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *srcReg = cg->evaluate(firstChild);
+   TR::Register *trgReg = cg->allocateRegister();
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic op;
 
-TR::Register *
-OMR::ARM64::TreeEvaluator::lrolEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::lrolEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   if (secondChild->getOpCodeValue() == TR::iconst || secondChild->getOpCodeValue() == TR::iuconst)
+      {
+      int32_t value = secondChild->getInt();
+      uint32_t shift = is64bit ? (value & 0x3F) : (value & 0x1F);
+
+      if (shift == 0)
+         {
+         generateMovInstruction(cg, node, trgReg, srcReg);
```

Comment:
This won't be necessary if `trgReg` == `srcReg` if my earlier suggestion is implemented.